tests/rsa_tests.rs

```diff
@@ -151,8 +151,8 @@ fn test_signature_rsa_pkcs1_verify() {
 
             let digest_name = test_case.consume_string("Digest");
             let alg = match digest_name.as_ref() {
-                "SHA1" => &signature::RSA_PKCS1_2048_8192_SHA1_FOR_LEGACY_USE_ONLY,
```

Comment:
I didn't look at the whole PR. However, I did look at this test. The test needs to be modified so that it verifies the correct results for each of `RSA_PKCS1_*` separately. For example, we need to verify that RSA_PKCS1_2048_8192_SHA256 still rejects the too-small inputs while RSA_PKCS1_1024_8192_SHA256 accepts the superset that are at least 1024 bits wide.

To accelerate the process of getting this all done, please fix this problem for `RSA_PKCS1_3072_8192_SHA384` too, which already had this problem. Otherwise, I'll do that myself before merging the PR.